src/Parser/DeathKnight/Unholy/Modules/Items/ColdHeartEfficiency.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import ITEMS from 'common/ITEMS';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class ColdHeartEfficiency extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  on_initialized() {
+    this.active = this.combatants.selected.hasChest(ITEMS.COLD_HEART.id);
+  }
+
+  totalColdHeartCasts = 0;
+  correctColdHeartCasts = 0;
+  buffColdHeart = 0;
+
+  on_byPlayer_applybuff(event){
+    this.addStack(event);
+  }
+
+  on_byPlayer_applybuffstack(event){
+    this.addStack(event);
+  }
+
+  addStack(event){
+    const spellID = event.ability.guid;
+    if(spellID === SPELLS.COLD_HEART_BUFF.id){
+      this.buffColdHeart+=1;
+	if(this.buffColdHeart > 20) {
+	this.buffColdHeart=20;
+	}
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId == SPELLS.CHAINS_OF_ICE.id) {
+       this.totalColdHeartCasts++;
+    	    if  (this.combatants.selected.hasBuff(SPELLS.UNHOLY_STRENGTH.id)) {
+		if (this.buffColdHeart < 20) {
+			if (this.buffColdHeart > 16) {
+       		        	this.correctColdHeartCasts++;
+				}
+			}
+                 }
+	    if (this.buffColdHeart == 20) {
+                 this.correctColdHeartCasts++;
+        }
+  }
+
+  }
```

Comment:
Looks like the logic is good.  Cleaning up the brackets will make it easier on the eyes of the admin who review this :)

Reply:
I'm sorry that it's so messy, I didn't know it would be included in the first pr I made.